src/commands/deploy/latest.js

```diff
@@ -518,36 +518,39 @@ export default async function main(
     if (!buildsCompleted) {
       const { builds: freshBuilds } = await now.fetch(buildsUrl);
 
-      for (const build of freshBuilds) {
-        const id = build.id;
-        const done = isDone(build);
+      // If there are no builds, we need to exit.
+      if (freshBuilds.length === 0) {
+        buildsCompleted = true;
+      } else {
+        for (const build of freshBuilds) {
+          const id = build.id;
+          const done = isDone(build);
 
-        if (times[id]) {
-          if (done && typeof times[id] === 'function') {
-            times[id] = times[id]();
+          if (times[id]) {
+            if (done && typeof times[id] === 'function') {
+              times[id] = times[id]();
+            }
+          } else {
+            times[id] = done ? allBuildsTime() : stamp();
           }
-        } else {
-          times[id] = done ? allBuildsTime() : stamp();
         }
-      }
 
-      if (JSON.stringify(builds) !== JSON.stringify(freshBuilds)) {
-        builds = freshBuilds;
+        if (JSON.stringify(builds) !== JSON.stringify(freshBuilds)) {
```

Comment:
Maybe we should use `assert.deepStrictEqual` here, since `JSON.stringify` doesn't work for properties that changed their orders.

Though, is probably not the case here

Reply:
Agreed.

But considering I only added a array length check earlier, I would not risk introducing another change that might break the system.